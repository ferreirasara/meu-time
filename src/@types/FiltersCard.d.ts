export type FiltersCardProps = {
  selectedCountry: string
  handleSelectCountry: (country: string) => void
  selectedLeague: number
  handleSelectLeague: (league: number) => void
  selectedSeason: number
  handleSelectSeason: (season: number) => void
  selectedTeam: number
  handleSelectTeam: (team: number) => void
}