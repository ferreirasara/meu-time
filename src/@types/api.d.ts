export type Team = {
  team: {
    id: number,
    name: string,
    code: string,
    country: string,
    founded: number,
    national: boolean,
    logo: string,
  },
  venue: {
    id: number,
    name: string,
    address: string,
    city: string,
    capacity: number,
    surface: string,
    image: string,
  },
}

export type Country = {
  name: string,
  code?: string,
  flag?: string,
}

export type League = {
  country: Country,
  league: {
    id: number,
    name: string,
    type: string,
    logo: string,
  }
  seasons: {
    year: number,
    start: string,
    end: string,
    current: boolean,
    coverage: {
      fixtures: {
        events: boolean,
        lineups: boolean,
        statistics_fixtures: boolean,
        statistics_players: boolean
      },
      standings: boolean,
      players: boolean,
      top_scorers: boolean,
      top_assists: boolean,
      top_cards: boolean,
      injuries: boolean,
      predictions: boolean,
      odds: boolean
    }
  }[],
}