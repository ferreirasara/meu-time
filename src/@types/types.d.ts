export type BaseData = {
  lastUpdated?: Date,
  countries?: Country[],
  leagues?: League[],
  seasons?: number[],
}

export type UserData = {
  email: string,
  firstname: string,
  lastname: string,
}

export type RequestData = {
  current: number,
  limitDay: number,
}

export type PlayersTableData = {
  id: number,
  name: string,
  age: number,
  nationality: string,
  photo: string,
}

export type PlayersTableProps = {
  selectedLeague: number
  selectedSeason: number
  selectedTeam: number
  handleError: (message: string) => void
}

export type GoalsChartProps = {
  data?: GeneralStatsData,
  loading: boolean,
}

export type GeneralStatsData = {
  totalDraws: number,
  totalPlayed: number,
  totalWins: number,
  totalLoses: number,
  mostUsedFormation: {
    formation: string,
    played: number,
  },
  goals: {
    against: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
    for: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
  }
}

export type GeneralStatsProps = {
  selectedLeague: number
  selectedSeason: number
  selectedTeam: number
  handleError: (message: string) => void
}

export type FiltersCardProps = {
  selectedCountry: string
  handleSelectCountry: (country: string) => void
  selectedLeague: number
  handleSelectLeague: (league: number) => void
  selectedSeason: number
  handleSelectSeason: (season: number) => void
  selectedTeam: number
  handleSelectTeam: (team: number) => void
  localStorageDataLoading: boolean
  handleError: (message: string) => void
}

export type Team = {
  team: {
    id: number,
    name: string,
    code: string,
    country: string,
    founded: number,
    national: boolean,
    logo: string,
  },
  venue: {
    id: number,
    name: string,
    address: string,
    city: string,
    capacity: number,
    surface: string,
    image: string,
  },
}

export type Country = {
  name: string,
  code?: string,
  flag?: string,
}

export type League = {
  country: Country,
  league: {
    id: number,
    name: string,
    type: string,
    logo: string,
  }
  seasons: {
    year: number,
    start: string,
    end: string,
    current: boolean,
    coverage: {
      fixtures: {
        events: boolean,
        lineups: boolean,
        statistics_fixtures: boolean,
        statistics_players: boolean
      },
      standings: boolean,
      players: boolean,
      top_scorers: boolean,
      top_assists: boolean,
      top_cards: boolean,
      injuries: boolean,
      predictions: boolean,
      odds: boolean
    }
  }[],
}

export type Player = {
  player: {
    id: number,
    name: string,
    firstname: string,
    lastname: string,
    age: number,
    birth: {
      date: string,
      place: string,
      country: string,
    },
    nationality: string,
    height: string,
    weight: string,
    injured: boolean,
    photo: string,
  },
  // statistics: {
  //   "team": {
  //     "id": 147,
  //     "name": "Coritiba",
  //     "logo": "https://media-1.api-sports.io/football/teams/147.png"
  //   },
  //   "league": {
  //     "id": 71,
  //     "name": "Serie A",
  //     "country": "Brazil",
  //     "logo": "https://media-2.api-sports.io/football/leagues/71.png",
  //     "flag": "https://media-3.api-sports.io/flags/br.svg",
  //     "season": 2016
  //   },
  //   "games": {
  //     "appearences": 6,
  //     "lineups": 6,
  //     "minutes": 438,
  //     "number": null,
  //     "position": "Defender",
  //     "rating": null,
  //     "captain": false
  //   },
  //   "substitutes": {
  //     "in": 0,
  //     "out": 3,
  //     "bench": 0
  //   },
  //   "shots": {
  //     "total": null,
  //     "on": null
  //   },
  //   "goals": {
  //     "total": 0,
  //     "conceded": null,
  //     "assists": null,
  //     "saves": null
  //   },
  //   "passes": {
  //     "total": null,
  //     "key": null,
  //     "accuracy": null
  //   },
  //   "tackles": {
  //     "total": null,
  //     "blocks": null,
  //     "interceptions": null
  //   },
  //   "duels": {
  //     "total": null,
  //     "won": null
  //   },
  //   "dribbles": {
  //     "attempts": null,
  //     "success": null,
  //     "past": null
  //   },
  //   "fouls": {
  //     "drawn": null,
  //     "committed": null
  //   },
  //   "cards": {
  //     "yellow": 0,
  //     "yellowred": 0,
  //     "red": 0
  //   },
  //   "penalty": {
  //     "won": null,
  //     "commited": null,
  //     "scored": null,
  //     "missed": null,
  //     "saved": null
  //   }
  // }[]
}

export type TeamStats = {
  lineups: {
    formation: string,
    played: number
  }[],
  fixtures: {
    draws: { home: number, away: number, total: number }
    loses: { home: number, away: number, total: number },
    played: { home: number, away: number, total: number },
    wins: { home: number, away: number, total: number },
  },
  goals: {
    against: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
    for: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
  }
}

export type InfoModalProps = {
  infoModalOpen: boolean
  onClose: () => void
}