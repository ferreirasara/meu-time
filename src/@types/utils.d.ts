export type BaseData = {
  lastUpdated?: Date,
  countries?: Country[],
  leagues?: League[],
  seasons?: number[],
}