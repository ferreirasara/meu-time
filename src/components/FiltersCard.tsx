import { Avatar, Card, Form, Select } from "antd"
import { getCountriesFromLocalStorage, getLeaguesFromLocalStorage, getSeasonsFromLocalStorage, getTeamsFromAPI } from "../utils/utils";
import { useEffect, useMemo, useState } from "react";
import { FiltersCardProps, Team } from "../@types/types";
import { LoadingOutlined } from "@ant-design/icons";
import useCheckMobileScreen from "../hooks/useCheckMobileScreen";

const getSuffixIcon = (src?: string, loading?: boolean) => {
  if (loading) {
    return <LoadingOutlined />
  } else if (src) {
    return <Avatar size="small" src={src} />
  } else {
    return null
  }
}

export const FiltersCard = ({
  selectedCountry,
  handleSelectCountry,
  selectedLeague,
  handleSelectLeague,
  selectedSeason,
  handleSelectSeason,
  selectedTeam,
  handleSelectTeam,
  localStorageDataLoading,
  handleError,
}: FiltersCardProps) => {
  const isMobile = useCheckMobileScreen();
  const [teams, setTeams] = useState<Team[]>([]);
  const [loadingTeams, setLoadingTeams] = useState<boolean>(false);

  const countries = getCountriesFromLocalStorage();
  const leagues = getLeaguesFromLocalStorage();
  const seasons = getSeasonsFromLocalStorage();

  const filteredLeagues = useMemo(() => leagues?.filter(league => league?.country?.name === selectedCountry), [leagues, selectedCountry]);

  const selectedCountryLogo = useMemo(() => countries?.find(cur => cur?.name === selectedCountry)?.flag, [countries, selectedCountry]);
  const selectedLeagueLogo = useMemo(() => filteredLeagues?.find(cur => cur?.league?.id === selectedLeague)?.league?.logo, [filteredLeagues, selectedLeague]);
  const selectedTeamLogo = useMemo(() => teams?.find(cur => cur?.team?.id === selectedTeam)?.team?.logo, [teams, selectedTeam]);

  useEffect(() => {
    if (selectedLeague && selectedSeason) {
      setLoadingTeams(true);
      getTeamsFromAPI(selectedLeague, selectedSeason)
        .then(result => setTeams(result))
        .catch(e => handleError(e?.toString()));
      setLoadingTeams(false);
    }
  }, [selectedLeague, selectedSeason, handleError])

  return <Card style={{ maxWidth: 1300 }}>
    <Form
      layout={isMobile ? "vertical" : "inline"}
    >
      <Form.Item
        label="País"
        name="country"
      >
        <Select
          placeholder="Selecione um país"
          showSearch
          options={countries?.map(country => ({ label: country?.name, value: country?.name }))}
          onChange={handleSelectCountry}
          value={selectedCountry}
          filterOption={(input, option) =>
            (option?.label ?? '').toLowerCase().includes(input.toLowerCase())
          }
          suffixIcon={getSuffixIcon(selectedCountryLogo, localStorageDataLoading)}
          style={{ width: isMobile ? "100%" : 150 }}
          loading={localStorageDataLoading}
          disabled={localStorageDataLoading}
          data-testid="select-country"
        />
      </Form.Item>

      <Form.Item
        label="Liga"
        name="league"
      >
        <Select
          disabled={!selectedCountry || localStorageDataLoading}
          placeholder="Selecione uma liga"
          showSearch
          options={filteredLeagues?.map(league => ({ label: league?.league?.name, value: league?.league?.id }))}
          onChange={handleSelectLeague}
          value={selectedLeague}
          filterOption={(input, option) =>
            (option?.label ?? '').toLowerCase().includes(input.toLowerCase())
          }
          suffixIcon={getSuffixIcon(selectedLeagueLogo, localStorageDataLoading)}
          style={{ width: isMobile ? "100%" : 150 }}
          loading={localStorageDataLoading}
          data-testid="select-league"
        />
      </Form.Item>

      <Form.Item
        label="Temporada"
        name="season"
      >
        <Select
          disabled={!selectedCountry || !selectedLeague || localStorageDataLoading}
          placeholder="Selecione uma temporada"
          showSearch
          options={seasons?.map(season => ({ label: season, value: season }))}
          onChange={handleSelectSeason}
          value={selectedSeason}
          style={{ width: isMobile ? "100%" : 150 }}
          loading={localStorageDataLoading}
          data-testid="select-season"
        />
      </Form.Item>

      <Form.Item
        label="Time"
        name="team"
      >
        <Select
          disabled={!selectedCountry || !selectedLeague || !selectedSeason || loadingTeams || localStorageDataLoading}
          placeholder="Selecione um time"
          showSearch
          loading={loadingTeams || localStorageDataLoading}
          options={teams?.map(team => ({ label: team?.team?.name, value: team?.team?.id }))}
          onChange={handleSelectTeam}
          value={selectedTeam}
          suffixIcon={getSuffixIcon(selectedTeamLogo, (loadingTeams || localStorageDataLoading))}
          style={{ width: isMobile ? "100%" : 150 }}
          data-testid="select-team"
        />
      </Form.Item>
    </Form>
  </Card>
}