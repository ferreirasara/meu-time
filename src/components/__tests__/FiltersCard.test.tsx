import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { FiltersCard } from '../FiltersCard';
import { FiltersCardProps } from '../../@types/types';

const props: FiltersCardProps = {
  selectedCountry: "",
  handleSelectCountry: jest.fn(),
  selectedLeague: 0,
  handleSelectLeague: jest.fn(),
  selectedSeason: 0,
  handleSelectSeason: jest.fn(),
  selectedTeam: 0,
  handleSelectTeam: jest.fn(),
  localStorageDataLoading: false,
  handleError: jest.fn(),
}

describe('FiltersCard', () => {
  beforeAll(() => {
    window.matchMedia = (query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })
  })

  it('match snapshot', () => {
    const filtersCard = renderer.create(<FiltersCard {...props} />).toJSON();
    expect(filtersCard).toMatchSnapshot();
  });

  it('renders all selectors', () => {
    const { getByTestId } = render(<FiltersCard {...props} />);

    const countrySelector = getByTestId("select-country");
    expect(countrySelector).toBeInTheDocument();

    const leagueSelector = getByTestId("select-league");
    expect(leagueSelector).toBeInTheDocument();

    const seasonSelector = getByTestId("select-season");
    expect(seasonSelector).toBeInTheDocument();

    const teamSelector = getByTestId("select-team");
    expect(teamSelector).toBeInTheDocument();
  });

  it('renders all labels', () => {
    const { getByText } = render(<FiltersCard {...props} />);

    const countryLabel = getByText(/Pa√≠s/, { exact: false });
    expect(countryLabel).toBeInTheDocument();

    const leagueLabel = getByText(/Liga/);
    expect(leagueLabel).toBeInTheDocument();

    const seasonLabel = getByText(/Temporada/);
    expect(seasonLabel).toBeInTheDocument();

    const teamLabel = getByText(/Time/);
    expect(teamLabel).toBeInTheDocument();
  });
})