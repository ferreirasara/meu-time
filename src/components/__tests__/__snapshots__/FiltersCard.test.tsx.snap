// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersCard match snapshot 1`] = `
<div
  className="ant-card ant-card-bordered css-dev-only-do-not-override-15rg2km"
  style={
    Object {
      "maxWidth": 1300,
    }
  }
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <form
      className="ant-form ant-form-inline css-dev-only-do-not-override-15rg2km"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="ant-form-item css-dev-only-do-not-override-15rg2km"
      >
        <div
          className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <label
              className=""
              htmlFor="country"
              title="País"
            >
              País
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <div
                  className="ant-select ant-select-in-form-item css-dev-only-do-not-override-15rg2km ant-select-single ant-select-show-arrow ant-select-show-search"
                  data-testid="select-country"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "width": 150,
                    }
                  }
                >
                  <div
                    className="ant-select-selector"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                  >
                    <span
                      className="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="country_list_0"
                        aria-autocomplete="list"
                        aria-controls="country_list"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        aria-owns="country_list"
                        autoComplete="off"
                        className="ant-select-selection-search-input"
                        disabled={false}
                        id="country"
                        onChange={[Function]}
                        onCompositionEnd={[Function]}
                        onCompositionStart={[Function]}
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onPaste={[Function]}
                        readOnly={false}
                        role="combobox"
                        style={
                          Object {
                            "opacity": null,
                          }
                        }
                        type="search"
                        unselectable={null}
                        value=""
                      />
                    </span>
                    <span
                      className="ant-select-selection-placeholder"
                    >
                      Selecione um país
                    </span>
                  </div>
                  <span
                    aria-hidden={true}
                    className="ant-select-arrow"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "WebkitUserSelect": "none",
                        "userSelect": "none",
                      }
                    }
                    unselectable="on"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-form-item css-dev-only-do-not-override-15rg2km"
      >
        <div
          className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <label
              className=""
              htmlFor="league"
              title="Liga"
            >
              Liga
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <div
                  className="ant-select ant-select-in-form-item css-dev-only-do-not-override-15rg2km ant-select-single ant-select-show-arrow ant-select-disabled ant-select-show-search"
                  data-testid="select-league"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "width": 150,
                    }
                  }
                >
                  <div
                    className="ant-select-selector"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                  >
                    <span
                      className="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="league_list_0"
                        aria-autocomplete="list"
                        aria-controls="league_list"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        aria-owns="league_list"
                        autoComplete="off"
                        className="ant-select-selection-search-input"
                        disabled={true}
                        id="league"
                        onChange={[Function]}
                        onCompositionEnd={[Function]}
                        onCompositionStart={[Function]}
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onPaste={[Function]}
                        readOnly={false}
                        role="combobox"
                        style={
                          Object {
                            "opacity": null,
                          }
                        }
                        type="search"
                        unselectable={null}
                        value=""
                      />
                    </span>
                    <span
                      className="ant-select-selection-placeholder"
                    >
                      Selecione uma liga
                    </span>
                  </div>
                  <span
                    aria-hidden={true}
                    className="ant-select-arrow"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "WebkitUserSelect": "none",
                        "userSelect": "none",
                      }
                    }
                    unselectable="on"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-form-item css-dev-only-do-not-override-15rg2km"
      >
        <div
          className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <label
              className=""
              htmlFor="season"
              title="Temporada"
            >
              Temporada
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <div
                  className="ant-select ant-select-in-form-item css-dev-only-do-not-override-15rg2km ant-select-single ant-select-show-arrow ant-select-disabled ant-select-show-search"
                  data-testid="select-season"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "width": 150,
                    }
                  }
                >
                  <div
                    className="ant-select-selector"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                  >
                    <span
                      className="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="season_list_0"
                        aria-autocomplete="list"
                        aria-controls="season_list"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        aria-owns="season_list"
                        autoComplete="off"
                        className="ant-select-selection-search-input"
                        disabled={true}
                        id="season"
                        onChange={[Function]}
                        onCompositionEnd={[Function]}
                        onCompositionStart={[Function]}
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onPaste={[Function]}
                        readOnly={false}
                        role="combobox"
                        style={
                          Object {
                            "opacity": null,
                          }
                        }
                        type="search"
                        unselectable={null}
                        value=""
                      />
                    </span>
                    <span
                      className="ant-select-selection-placeholder"
                    >
                      Selecione uma temporada
                    </span>
                  </div>
                  <span
                    aria-hidden={true}
                    className="ant-select-arrow"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "WebkitUserSelect": "none",
                        "userSelect": "none",
                      }
                    }
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      className="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-form-item css-dev-only-do-not-override-15rg2km"
      >
        <div
          className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <label
              className=""
              htmlFor="team"
              title="Time"
            >
              Time
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <div
                  className="ant-select ant-select-in-form-item css-dev-only-do-not-override-15rg2km ant-select-single ant-select-show-arrow ant-select-disabled ant-select-show-search"
                  data-testid="select-team"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "width": 150,
                    }
                  }
                >
                  <div
                    className="ant-select-selector"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                  >
                    <span
                      className="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="team_list_0"
                        aria-autocomplete="list"
                        aria-controls="team_list"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        aria-owns="team_list"
                        autoComplete="off"
                        className="ant-select-selection-search-input"
                        disabled={true}
                        id="team"
                        onChange={[Function]}
                        onCompositionEnd={[Function]}
                        onCompositionStart={[Function]}
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onPaste={[Function]}
                        readOnly={false}
                        role="combobox"
                        style={
                          Object {
                            "opacity": null,
                          }
                        }
                        type="search"
                        unselectable={null}
                        value=""
                      />
                    </span>
                    <span
                      className="ant-select-selection-placeholder"
                    >
                      Selecione um time
                    </span>
                  </div>
                  <span
                    aria-hidden={true}
                    className="ant-select-arrow"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "WebkitUserSelect": "none",
                        "userSelect": "none",
                      }
                    }
                    unselectable="on"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
